### PR TITLE
WIP: Suggest duplicates

### PR DESCRIPTION
**Pull request by symroe**
Connects #317 

Will add more when it's no longer WIP.

### CODE REVIEW

File: ynr/apps/duplicates/models.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+    """
+    A model for storing duplicate suggestions.
+
+    TODO:
+
+    * If a person has N duplicate suggestions and one gets merged in to the
+      other, what do we do?
+    * Mark people known as not duplicate in a KnownNotDuplicateQuerySet –
+      this prevents people from marking someone as a duplicate when it's known
+      they are not (for example, because they've been suggested before)
```
Comment (chris48s) [inline ynr/apps/duplicates/models.py]: It seems like another useful thing here would be for the model to be aware that a duplicate suggestion for
`(person=1, other_person=2)` and a duplicate suggestion for `(person=2, other_person=1)` are effectively the same record so we don't get.. duplicate duplicates :) Maybe do that check on `save()`?

**Comment by symroe** [comment]
This is looking great!

It looks like there are a few things still do to (going to list them just in case you don't have them on your TODO yet):

## Blockers to merging

* [ ] Linking to this in the menu for people who have merge permissions (with a number, hiding if none exist)
* [ ] Add the person ID to the person pages somewhere


## Nice to have / to do after deploying MVP (to be made in to an issue I expect, but let's dump here for now)

* [ ] UX Improvements to the initial review screen - adding a table to put details side by side at least
* [ ] I think the page that shows when you suggest a duplicate that's previously been rejected should show the reason given
* [ ] Some small improvements to the inline rejection form: I think we should make the input a text area (the reasons might be long!) and only show it when someone presses the reject button, like we do with the "not actually standing" buttons on the bulk adding forms
* [ ] Hide the merge form if there is already a merge suggestion for this person

**Comment by michaeljcollinsuk** [comment]
Re: blockers

- Person ID added here to the main profile page here https://github.com/DemocracyClub/yournextrepresentative/pull/442/files#diff-7ae1e99d65bad589ccee1d1e5877a0fb29a76b6f0a0dede511cb322bc860ed05R110
- Just pushed commit to add notification and link to the header

**Comment by symroe** [comment]
> Person ID added here to the main profile page here
Ah, sorry I'd missed this :+1:

**Comment by michaeljcollinsuk** [comment]
One other change I just remembered I wanted to check you were happy with is [this commit](https://github.com/DemocracyClub/yournextrepresentative/pull/442/commits/0c9eae007aa22eba5fe0bcaea419b8a0b16ded48) where I default `versions` to a list rather than allowing the field to be `null`. As there were errors occurring attempting to merge a person with versions, with someone that didnt have any. I grepped the code and couldn't find any knock on effects of this change, but in case you merge this weekend just wanted to draw your attention to it.

**Comment by michaeljcollinsuk** [comment]
@symroe one other change I just remembered I wanted to check you were happy with is [this commit](https://github.com/DemocracyClub/yournextrepresentative/pull/442/commits/0c9eae007aa22eba5fe0bcaea419b8a0b16ded48) where I default `versions` to a list rather than allowing the field to be `null`. As a `TypeError` was being raised when [attempting to merge a person with versions, with someone that didnt have any](https://github.com/DemocracyClub/yournextrepresentative/blob/a6e768ec64b23eaede3eba7f021a86e741b70598/ynr/apps/people/merging.py#L154). I grepped the code and couldn't find any knock on effects of this change. But just wanted to draw your attention to it in case you merge this weekend.

**Pull request merged**